external/js/cothority/src/network/connection.ts

```diff
@@ -191,67 +188,66 @@ export class WebSocketConnection implements IConnection {
     }
 
     /** @inheritdoc */
-    sendStream<T extends Message>(message: Message, reply: typeof Message,
-                                  onMessage: (data: T, ws: WebSocketAdapter) => void,
-                                  onClose: (code: number, reason: string) => void,
-                                  onError: (err: Error) => void): WebSocketAdapter {
+    sendStream<T extends Message>(message: Message, reply: typeof Message):
+        Observable<[T, WebSocketAdapter]> {
+        return new Observable((sub) => {
 
-        if (!message.$type) {
-            onError(new Error(`message "${message.constructor.name}" is not registered`));
-            return;
-        }
+            if (!message.$type) {
+                sub.error(new Error(`message "${message.constructor.name}" is not registered`));
```

Comment:
I would move it outside of the subscription, as it's an argument checking error, not a transient one.